SignalEducation/version1
# (AP-68) Login api.

What? Login endpoint.
How? api call to do user authentication.
How to test?

API call: I did add a user register test(not working yet, but you can test the API authentication using it.)
GET request to http://localhost:3000/api/v1/auth/login

Use Bearer Token authentication and add your API key as a value.

params:
```
{
    "user": {
        "email": "email@test.com",
        "password": "user_password"
    }
}
```

This call should return to you the user information + token that we'll use for the next authenticated calls.

If you want to test an authenticated call, I added a comment [before action](https://github.com/SignalEducation/version1/blob/b09b2fc37ed5c5e4851e314b337501719e9d4615/app/controllers/api/v1/users_controller.rb#L6) in our user API.

You can just uncomment this callback and use the endpoint passing the token returned from login as header param as you can see in the screenshots below.

![Screenshot 2021-08-11 at 10 13 04](https://user-images.githubusercontent.com/136358/129002965-0ca54f9a-88b6-4466-9fb3-976a65ae2b1d.png)
![Screenshot 2021-08-11 at 10 13 15](https://user-images.githubusercontent.com/136358/129002974-7689591a-64c0-4872-b27b-5ba77b04efd9.png)
